Entities/Items/Food/Eatable.as

```diff
@@ -78,20 +97,43 @@ void onCollision(CBlob@ this, CBlob@ blob, bool solid)
 	}
 }
 
-
 void onAttach(CBlob@ this, CBlob@ attached, AttachmentPoint @attachedPoint)
 {
-	if (this is null || attached is null) {return;}
-
-	if (isServer())
-	{
-		Heal(attached, this);
-	}
-
-	CPlayer@ p = attached.getPlayer();
-	if (p is null){return;}
+    if (this is null || attached is null) {return;}
+
+    if (isServer())
+    {
+        Heal(attached, this);
+    }
+    
+    CPlayer@ player = attached.getPlayer();
+
+    if (player !is null && (this.getTeamNum() != attached.getTeamNum() || !this.exists("healer")))
+    {
+        if (isServer())
+        {
+            this.server_setTeamNum(attached.getTeamNum());
+        }
+
+        this.set_u16("healer", player.getNetworkID());
+        this.Sync("healer", true);
+    }
+}
 
-	this.set_u16("healer", p.getNetworkID());
+void onThisAddToInventory(CBlob@ this, CBlob@ inventoryBlob)
+{
+    CPlayer@ player = inventoryBlob.getPlayer();
+
+    if (player !is null && (this.getTeamNum() != inventoryBlob.getTeamNum() || !this.exists("healer")))
```

Comment:
That `if` condition logic is duplicated with the `onAttach` function, make a function for this, e.g. `void setHealer(CBlob@ food_blob, CPlayer@ player)` so you can call it like `setHealer(@this, @player);`

Reply:
added setHealer to EatCommon.as